### PR TITLE
Implement --convert-repeated-to-block option in nnet3-am-copy.

### PR DESCRIPTION
**Pull request by galv**
Addresses #440.

In addition to the unit tests provided, I ran nnet3-am-copy --repeated-to-block on one of my own models, and check all the components were successfully converted.

### CODE REVIEW

File: src/nnet3/nnet-nnet.h [hunk 1]
```diff
@@ -130,6 +130,12 @@ class Nnet {
   /// caller.
   const Component *GetComponent(int32 c) const;
 
+  /// Replace the component indexed by c with a new component.
+  /// Frees previous component indexed by c.
+  /// Changing the identity of a component is dangerous; this was made
+  /// only to replaced RepeatedAffine components with BlockAffine components,
```
Comment (danpovey) [inline src/nnet3/nnet-nnet.h]: You don't have to mention the reason it was added- it's a reasonable thing to have there regardless.

File: src/nnet3/nnet-simple-component.cc [hunk 2]
```diff
@@ -1369,6 +1369,26 @@ BlockAffineComponent::BlockAffineComponent(const BlockAffineComponent &other) :
   bias_params_(other.bias_params_),
   num_blocks_(other.num_blocks_) {}
 
+BlockAffineComponent::BlockAffineComponent(const RepeatedAffineComponent &rac) :
+  UpdatableComponent(rac),
+  linear_params_(rac.num_repeats_ * rac.linear_params_.NumRows(), 
+                 rac.linear_params_.NumCols(), kUndefined),
+  bias_params_(rac.num_repeats_ * rac.linear_params_.NumRows(), kUndefined),
+  num_blocks_(rac.num_repeats_)
+{
```
Comment (danpovey) [inline src/nnet3/nnet-simple-component.cc]: i think this brace should be at the end of the previous line.

File: src/nnet3/nnet-utils-test.cc [hunk 3]
```diff
@@ -45,6 +47,59 @@ void UnitTestNnetContext() {
   }
 }
 
+void UnitTestConvertRepeatedToBlockAffine() {
```
Comment (danpovey) [inline src/nnet3/nnet-utils-test.cc]: it's good that you added testing code.

File: src/nnet3bin/nnet3-am-copy.cc [hunk 4]
```diff
@@ -89,6 +95,10 @@ int main(int argc, char *argv[]) {
       am_nnet.SetNnet(nnet);
     }
 
+    if(convert_repeated_to_block) {
```
Comment (danpovey) [inline src/nnet3bin/nnet3-am-copy.cc]: braces not needed

**Comment by galv** [comment]
I have fixed the style problems.

File: src/nnet3/nnet-nnet.h [hunk 5]
```diff
@@ -130,6 +130,9 @@ class Nnet {
   /// caller.
   const Component *GetComponent(int32 c) const;
 
+  /// Replace the component indexed by c with a new component.
+  /// Frees previous component indexed by c.
```
Comment (danpovey) [inline src/nnet3/nnet-nnet.h]: Please mention that it takes ownership of the pointer 'component'.

**Pull request merged**